packages/nbmodel/src/ymodel.ts

```diff
@@ -459,6 +466,15 @@ export class YBaseCell<Metadata extends nbmodel.ISharedBaseCellMetada>
         newValue: this.getMetadata()
       };
     }
+
+    if (modelEvent && modelEvent.keysChanged.has('execution_count')) {
+      const change = modelEvent.changes.keys.get('execution_count');
+      changes.executionCountChange = {
+        oldValue: change?.oldValue ? change!.oldValue : undefined,
+        newValue: this.ymodel.get('execution_count')
+      };
```

Comment:
This change-request is not important. But I think the following style would be more elegant:

```js
if (modelEvent && modelEvent.keysChanged.has('execution_count')) {
      const change = modelEvent.changes.keys.get('execution_count')!;
      changes.executionCountChange = {
        oldValue: change.oldValue,
        newValue: this.ymodel.get('execution_count')
      };
```

Note that I simply added a `!` after `get('execution_count')` because we checked before that it exists.

Reply:
TypeScript doesn't know we check because we are accessing a different property.

Reply:
Right. But we know. If you add the exclamation mark `!` you tell typescript that the value must exist.